enterprise/backend/src/metabase_enterprise/metabot_v3/client.clj

```diff
@@ -129,6 +134,57 @@
     (catch Throwable e
       (throw (ex-info (format "Error in request to AI Proxy: %s" (ex-message e)) {} e)))))
 
+(mu/defn streaming-request :- :any
+  "Make a V2 request to the AI Proxy."
+  [{:keys [context messages profile-id conversation-id session-id state]}
+   :- [:map
+       [:context :map]
+       [:messages ::metabot-v3.client.schema/messages]
+       [:profile-id :string]
+       [:conversation-id :string]
+       [:session-id :string]
+       [:state :map]]]
+  (premium-features/assert-has-feature :metabot-v3 "MetaBot")
+  (try
+    (let [url      (agent-v2-streaming-endpoint-url)
+          body     (-> {:messages        messages
+                        :context         context
+                        :conversation_id conversation-id
+                        :profile_id      profile-id
+                        :state           state
+                        :user_id         api/*current-user-id*}
+                       (metabot-v3.u/recursive-update-keys metabot-v3.u/safe->snake_case_en))
+          _        (metabot-v3.context/log body :llm.log/be->llm)
+          _        (log/debugf "V2 request to AI Proxy:\n%s" (u/pprint-to-str body))
+          options  (cond-> {:headers          {"Accept"                    "text/event-stream"
+                                               "Content-Type"              "application/json;charset=UTF-8"
+                                               "x-metabase-instance-token" (premium-features/premium-embedding-token)
+                                               "x-metabase-session-token"  session-id
+                                               "x-metabase-url"            (system/site-url)}
+                            :body             (->json-bytes body)
+                            :throw-exceptions false
+                            :as :stream}
+                     *debug* (assoc :debug true))
+          response (post! url options)
+          response-lines ^java.io.BufferedReader (-> response :body io/reader)]
+      (metabot-v3.context/log (:body response) :llm.log/llm->be)
+      (log/debugf "Response from AI Proxy:\n%s" (u/pprint-to-str (select-keys response #{:body :status :headers})))
+      (if (= (:status response) 200)
+        (sr/streaming-response {:content-type "text/event-stream; charset=utf-8"
+                                :headers {"x-metabot-conversation-id" conversation-id}} [os canceled-chan]
+          (loop []
+            (when-let [line (.readLine response-lines)]
+              (.write os (.getBytes (str line "\n") "UTF-8"))
+              (.flush os)
```

Comment:
Are you sure you want to be flushing on every single line? Let the `streaming-response` code decide when to buffer and when to flush.